components/Repositories/AddToWorkspaceDrawer.tsx

```diff
@@ -51,15 +56,38 @@ export default function AddToWorkspaceDrawer({ repository }: { repository: strin
       toast({ description: `Added repository successfully`, variant: "success" });
     }
   };
+
+  const params = new URLSearchParams();
+
+  if (type === "sbom") {
+    params.set("sbom", "true");
+    params.set("repo", repository);
+  } else {
+    params.set("repos", JSON.stringify([repository]));
+  }
+
+  const url = `/workspaces/new?${params}`;
+  const redirectTo = new URL(url, window.location.origin).toString();
+  const title = type === "repo" ? "Add repository to Workspace" : "Add repository SBOM to Workspace";
+
   return (
     <Drawer
-      title="Add repository to Workspace"
+      title={title}
       description="Create a new workspace or add to an existing one."
       showCloseButton
       trigger={
-        <Button variant="primary" className="shrink-0 items-center gap-3 w-fit">
+        <Button
+          variant={type === "repo" ? "primary" : "dark"}
+          className="shrink-0 items-center gap-3 w-full"
+          onClick={(event) => {
+            if (user && type === "sbom") {
+              event.preventDefault();
```

Comment:
preventing default here so the trigger doesn't open the drawer since the user is already logged in.